File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    result = []
    answer = []
    
    for item in score:
        if len(answer) == 0:
            result.append(item)
        else:
            if len(answer) < k:
                result.append(item)
            else:
                if result[0] < item:
                    result[0] = item
        result.sort()
        answer.append(result[0])
    
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const currArray = [];
    const answer = score.map((item, index) => {
        currArray.push(item);
        if (currArray.length <= k) return Math.min(...currArray);
        return currArray.sort((a, b) => b - a)[k - 1];
    });
    return answer;
}